src/Microsoft.AspNet.OData.Shared/Formatter/ODataOutputFormatterHelper.cs

```diff
@@ -349,5 +348,65 @@ private static bool IsOperationPath(ODataPath path)
 
             return false;
         }
+
+        private static Microsoft.OData.UriParser.ODataPath ConvertPath(ODataPath path)
+        {
+            if (path == null)
+            {
+                return null;
+            }
+
+            if (IsOperationPath(path))
+            {
+                var lastSegment = path.Segments.Last();
+                OperationSegment operation = lastSegment as OperationSegment;
+                if (operation != null && operation.EntitySet != null)
+                {
+                    return GeneratePath(operation.EntitySet);
+                }
+
+                OperationImportSegment operationImport = lastSegment as OperationImportSegment;
+                if (operationImport != null && operationImport.EntitySet != null)
+                {
+                    return GeneratePath(operationImport.EntitySet);
+                }
+
+                return null;
+            }
+
+            return path.Path;
+        }
+
+        private static Microsoft.OData.UriParser.ODataPath GeneratePath(IEdmNavigationSource navigationSource)
+        {
+            switch (navigationSource.NavigationSourceKind())
+            {
+                case EdmNavigationSourceKind.EntitySet:
+                    return new Microsoft.OData.UriParser.ODataPath(new EntitySetSegment((IEdmEntitySet)navigationSource));
+
+                case EdmNavigationSourceKind.Singleton:
+                    return new Microsoft.OData.UriParser.ODataPath(new SingletonSegment((IEdmSingleton)navigationSource));
+
+                case EdmNavigationSourceKind.ContainedEntitySet:
+                    IEdmContainedEntitySet containedEntitySet = (IEdmContainedEntitySet)navigationSource;
+                    var path = GeneratePath(containedEntitySet.ParentNavigationSource);
```

Comment:
>var [](start = 20, length = 3)

better not to use var

Reply:
Yes. I know. I originally thought it's easy to use "var" not "Microsoft.OData.UriParser.ODataPath". But I changed.

---
In reply to: [386706993](https://github.com/OData/WebApi/pull/2060#discussion_r386706993) [](ancestors = 386706993)